scripts/validate_docstrings.py

```diff
@@ -331,6 +336,33 @@ def parameter_mismatches(self):
 
         return errs
 
+    @contextlib.contextmanager
+    def _write_examples_code_to_temp_file(self):
+        """
+        Generate file with source code from examples section.
+        """
+        content = ''.join(('import numpy as np; '
+                           'import pandas as pd  # noqa: F401,E702\n',
+                           *self.examples_source_code))
+        with tempfile.NamedTemporaryFile(mode='w') as file:
+            file.write(content)
+            file.flush()
+            yield file
+
+    def validate_pep8(self):
+        if self.examples:
+            with self._write_examples_code_to_temp_file() as file:
+                application = flake8.main.application.Application()
+                application.initialize(["--quiet"])
+                application.run_checks([file.name])
+                application.report()
```

Comment:
Sorry I wasn't clear in my previous review about this. This is what I'd do:
```
application = flake8.main.application.Application()
application.initialize(["--quiet"])
with tempfile.NamedTemporaryFile(mode='w') as f:
    f.write(content)
    f.flush()
    application.run_checks([f.name])
application.report()
```

I think it's a good practice to have inside the context manager, the code that requires to be there.

Also, as this file is going quite significantly, I think it's better to keep all this functionality in a single function, so each validation is somehow isolated. I think in another context it makes more sense what you did of splitting, but I don't think we'll reuse the temporary file creation, and I think the improvement in simplicity makes it worth.